Added Bamboo Documentation (#5871)
webdriverio/webdriverio: docs/BambooIntegration.md

---
id: bamboo
title: Bamboo Integration
---

> WebdriverIO offers a tight integration to CI systems like [Bamboo](https://www.atlassian.com/software/bamboo). With the `junit | allure` reporters, you can easily debug your tests as well as keep track of your test results. The integration is pretty easy.

Reviewer: How about we link the reporter:
```suggestion
WebdriverIO offers a tight integration to CI systems like [Bamboo](https://www.atlassian.com/software/bamboo). With the [JUnit](https://webdriver.io/docs/junit-reporter.html) or [Allure](https://webdriver.io/docs/allure-reporter.html) reporter, you can easily debug your tests as well as keep track of your test results. The integration is pretty easy.
```

> 1. Install the `junit` test reporter: `$ npm install @wdio/junit-reporter --save-dev`)

Reviewer: Let's stay consistent and call it JUnit
```suggestion
1. Install the JUnit test reporter: `$ npm install @wdio/junit-reporter --save-dev`)
```

> 1. Update your config to save your XUnit results where Bamboo can find them, (and specify the `junit` reporter):

Reviewer: ```suggestion
1. Update your config to save your JUnit results where Bamboo can find them, (and specify the `junit` reporter):
```

```js
// wdio.conf.js
module.exports = {
    // ...
    reporters: [
        'dot',
        ['junit', {
            outputDir: './testresults/'
        }]
    ],
    // ...
}
```
Note: *It's always a good standard to keep the test results in seperate folder than in the root folder.*

```js
// wdio.conf.js - For tests running in parallel
module.exports = {
    // ...
    reporters: [
        'dot',
        ['junit', {
            outputDir: './testresults/'
        },
        outputFileFormat: function (options) {
            return `results-${options.cid}.xml`;
        }]
    ],
    // ...
}
```

> The reports will be similar for all the frameworks and you can use anyone ```Mocha | Jasmine | Cucumber```.

Reviewer: ```suggestion
The reports will be similar for all the frameworks and you can use anyone: Mocha, Jasmine or Cucumber.
```

By this time, we believe you have the tests written up and results are generated in ```./testresults/``` folder, and your Bamboo is up and running.

## Integrate your tests in Bamboo

1. Open your Bamboo project
    > Create a new plan, link your repository (make sure it always points to newest version of your repository) and create your stages

    ![Plan Details](/img/bamboo/plancreation.png "Plan Details")

    I will go with the default stage and job. In your case, you can create your own stages and jobs

    ![Default Stage](/img/bamboo/defaultstage.png "Default Stage")

> 
> 
> 

Reviewer: ```suggestion

```
